### ISSUE TITLE
Use redux to handle authentication state

### ISSUE THREAD

**Issue by batflarrow** [iiitv/CTF-frontend#7]

**Comment by toth2000** [comment]
I can look into it.

**Comment by batflarrow** [comment]
> I can look into it.

Basically what needs to be done is when a guy logs in i.e. login API is called set the authentication state to true and after this whenever a guy refreshes a api will be called named /authContinuity which will verify if you are logged in or not if the response of this api is not authenticated set the state to false. Don't think about the exact responses you can assume the status codes to be 200 for logged in and 401 for not authenticated. So are you up for it..?

**Comment by toth2000** [comment]
Okay, so I have to create a Redux store that will hold a boolean value of the login state.

**Comment by batflarrow** [comment]
> Okay, so I have to create a Redux store that will hold a boolean value of the login state?.

Yeah sure a redux store that will be accessible through the application and store the login state.

**Comment by raghusodani** [comment]
anyone working on this @batflarrow ??

**Comment by batflarrow** [comment]
Naah all yours f you are up for it.. @raghusodani

**Comment by raghusodani** [comment]
ok then assign me

**Comment by batflarrow** [comment]
Go for it @raghusodani

**Comment by raghusodani** [comment]
there are no login data till now so I am framing a generic reducer to handle username and email.
is it fine?

**Comment by batflarrow** [comment]
> there are no login data till now so I am framing a generic reducer to handle username and email. is it fine?

Yeah sure username and email will do just fine and a boolean isAuthenticated so as to maintain the state of auth

**Comment by raghusodani** [comment]
> > there are no login data till now so I am framing a generic reducer to handle username and email. is it fine?
> 
> Yeah sure username and email will do just fine and a boolean isAuthenticated so as to maintain the state of auth

sure
